src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,214 @@
+                + "WHERE c.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCarAndManufacturerFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to get all cars from DB", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversForCar(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars "
+                + "SET manufacturer_id = ?, model = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(FIRST_INDEX, car.getManufacturer().getId());
+            statement.setString(SECOND_INDEX, car.getModel());
+            statement.setLong(THIRD_INDEX, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to update " + car, e);
+        }
+        deleteAllCarDrivers(car);
+        insertCarDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars "
+                + "SET is_deleted = TRUE "
+                + "WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(FIRST_INDEX, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to delete car. Id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Long> carIDs = new ArrayList<>();
+        String query = "SELECT car_id "
+                + "FROM cars_drivers "
+                + "WHERE driver_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(FIRST_INDEX, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                carIDs.add(resultSet.getLong("car_id"));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to get all "
+                    + "car IDs with driver that has ID:" + driverId, e);
+        }
+        List<Car> cars = new ArrayList<>();
+        for (Long id : carIDs) {
+            get(id).ifPresent(cars::add);
+        }

```

Comment:
can be refactored in functional style with using `stream ` and `filter` instead of using `ifPresent()`